core/store/tx_manager.go

```diff
@@ -238,22 +238,28 @@ func (txm *EthTxManager) createTx(
 	gasLimit uint64,
 	value *assets.Eth) (*models.Tx, error) {
 
-	tx, err := txm.sendInitialTx(surrogateID, ma, to, data, gasPriceWei, gasLimit, value)
-	for nrc := 0; isNonceTooLowError(err); nrc++ {
-		logger.Warnw("Tx #0: nonce too low, retrying with network nonce")
+	for nrc := 0; nrc <= nonceReloadLimit; nrc++ {
+		tx, err := txm.sendInitialTx(surrogateID, ma, to, data, gasPriceWei, gasLimit, value)
+		if err == nil {
+			return tx, nil
+		}
 
-		if nrc >= nonceReloadLimit {
-			err = fmt.Errorf(
-				"Transaction reattempt limit reached for 'nonce is too low' error. Limit: %v",
-				nonceReloadLimit,
-			)
-			return nil, err
+		if !isNonceTooLowError(err) {
+			return nil, errors.Wrap(err, "TxManager#retryInitialTx sendInitialTx")
 		}
 
-		err = txm.retryInitialTx(tx, ma, gasPriceWei)
+		logger.Warnw("Tx #0: nonce too low, retrying with network nonce")
+
+		err = ma.ReloadNonce(txm)
+		if err != nil {
+			return nil, errors.Wrap(err, "TxManager#retryInitialTx ReloadNonce")
+		}
 	}
 
-	return tx, err
+	return nil, fmt.Errorf(
+		"Transaction reattempt limit reached for 'nonce is too low' error. Limit: %v",
+		nonceReloadLimit,
+	)
 }
 
 // sendInitialTx creates the initial Tx record + attempt for an Ethereum Tx,
```

Comment:
Looks like this comment is out of date, this function is being called on every single tx attempt now.
Maybe we should rename it too, to something like `sendTx`

Reply:
That comment looks right to me, this code is only being used to send the initial transaction, though it is called in a loop because the initial transaction is retried while we see a `nonce too low` error.

Reply:
Hm, to me, if the transaction hash changes at all (which it would if you change something like nonce), then it is no longer the same transaction.

Reply:
The only thing that seems to be staying static (at a glance) is the `data` and `to` field of the transaction

Reply:
Correct, but not transaction will have been accepted yet, so it's still going to try and send the initial transaction, until the initial transaction can be saved. So this will always create a TxAttempt in position 0.